Library/Homebrew/cmd/--prefix.rb

```diff
@@ -39,7 +43,13 @@ def __prefix
       puts HOMEBREW_PREFIX
     else
       puts args.named.to_resolved_formulae.map { |f|
-        f.opt_prefix.exist? ? f.opt_prefix : f.latest_installed_prefix
+        if f.opt_prefix.exist?
+          f.opt_prefix
+        elsif args.installed?
+          ""
```

Comment:
```suggestion
          Homebrew.failed = true
          ""
```

to set the exit code to non-0

Reply:
Hmmm. Now that I've had a second think, I can see the logic of returning failure, given the name of the new option.